### DISCUSSION TITLE
Some Luxon zone offsets seem to be incorrect

### DISCUSSION THREAD

**Discussion by alberthtan** [moment/luxon#1478] [Q&A] [answered]
Hey! When I'm using Luxon timezones, some zone offsets seem to be incorrect. Ex. Singapore returns +7:30, even though it is UTC + 8:00 (last time it was +7:30 was in 1981). Other examples include:

Turkey - returns +2 but is actually +3
Pacific/Norfolk - displays +11:30 but is +11
Pacific/Kiritimati - displays -10:40 but is +14

Is there any fix / support for this?

**Comment by diesieben07** [comment]
Luxon gets its time zone (and other locale) data from the underlying platform via the `Intl.DateTimeFormat` API. Do you have more information about the environment you are using, i.e. which browser/operating system/etc.?

**Reply by alberthtan** [reply]
I'm using Chrome and MacOS. For example, when I do this:

luxonDateObject.setZone("Singapore")

console.log(luxonDateObject.zone.formatOffset(
  now.millisecond,
  "short"
}); 

// returns +7:30 even though the correct one is +8:00
[singapore time zone](https://en.wikipedia.org/wiki/Singapore_Standard_Time#:~:text=Singapore%20Standard%20Time%20(SST)%2C,not%20observe%20daylight%20saving%20time.&text=DST%20is%20not%20observed%20in%20this%20time%20zone.)

**Reply by diesieben07** [reply, marked as answer]
`now.millisecond` is a value between 0 and 999 denoting the fractional milliseconds of the `DateTime`. `Zone#formatOffset` expects a unix timestamp in milliseconds. Therefor your code will always format the offset on some fractional offset within the first second of 1970/01/01. At this point `Asia/Singapore` was at offset +7:30, the zone was only shifted to +8:00 in 1982.
If you pass the correct timestamp to `formatOffset` you will receive the correct offset.

**Edit:** to get the unix timestamp in millseconds from a `DateTime` use `DateTime#valueOf` or `DateTime#toMillis`.

**Edit again:** I jused noticed that in your code you are using `luxonDateObject.setZone("Singapore")` without using the return value. Luxon's objects are immutable, therefor this call as shown has no effect.

**Reply by alberthtan** [reply]
Thanks so much! I'll try to figure out a solution.

Yes, I sort of abbreviated it for the context of the question, but I was doing something like "now = luxonDateObject.setZone..."

**Reply by alberthtan** [reply]
Thanks. Just resolved the issue with DateTime#toMillis!
